text/0000-builder-spec.md

```diff
@@ -0,0 +1,270 @@
+* The image config's Label field has the label io.buildpacks.buildpack.layers, set to a JSON object representing the [layers](#layers)
+
+#### Order
+The `io.buildpacks.buildpack.order` data should look like:
+```json
+[
+	{
+	  "group":
+		[
+		  {
+			"id": "<buildpack id>",
+			"version": "<buildpack version>",
+			"optional": "<bool>"
+		  }
+		]
+	}
+]
+```
+
+#### Metadata
+The `io.buildpacks.builder.metadata` data should look like:
+```json
+{
+  "description": "<Some description>",
+  "stack": {
+    "runImage": {
+      "image": "<some/run-image>",
+      "mirrors": [
+        "<gcr.io/some/default>"
+      ]
+    }
+  },
+  "buildpacks": [
+    {
+      "id": "<buildpack id>",
+	  "version": "<buildpack version>",
+	  "homepage": "http://geocities.com/top-bp"
+	}
+  ],
+  "lifecycle": {
+    "version": "<version of lifecycle in builder>",
+    "apis":  {
+        "buildpack": {
+          "deprecated": ["0.1"],
+          "supported": ["1.2", "1.3"]},
+        "platform": {
+          "deprecated": [],
+          "supported": ["2.3", "2.4"]}
+    }
+  },
+  "createdBy": {
+    "name": "<creator of builder>", 
+    "version": "<builder version>"
+  }
+}
+```
+
+#### Layers
+The `io.buildpacks.buildpack.layers` data should look like:
+```json
+{
+  "<buildpack id>": {
+    "<buildpack version>": {
+      "api": "<buildpack API>",
+      "order": [
+        {
+          "group": [
+            {
+              "id": "<inner buildpack>",
+              "version": "<inner bulidpacks version>"
+            }
+          ]
+        }
+      ],
+      "layerDiffID": "sha256:test.nested.sha256",
+	  "homepage": "http://geocities.com/top-bp"
+    }
+  },
+  "<inner buildpack>": {
+    "<inner buildpacks version>": {
+      "api": "<buildpack API>",
+      "stacks": [
+        {
+          "id": "<stack ID buildpack supports>",
+          "mixins": ["<list of mixins required>"]
+        }
+      ],
+      "layerDiffID": "sha256:test.bp.one.sha256",
+	  "homepage": "http://geocities.com/cool-bp"
+    }
+  }
+}
+```
+
+## Spec Additions
+### [Buildpack specification][buildpacks-spec]
+Builders don't seem immediately relevant to the buildpack specification; as such, I wouldn't add mention of builders here. 
+
+### [Platform specification][platform-spec]
+We can add a note, clarifying that in platforms that use builders, the Stack's build image is the core of the builder. 
+
+### [Distribution specification][distribution-spec]
+We should have a rigorous definition of the builder here, with a description of all the files and metadata necessary and allowed. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+N/A
+
+# Alternatives
+[alternatives]: #alternatives
+Specifying the builder is something that, to my mind, should definitely done; the question is, how. We could have an entirely separate `extension spec`, defining builders and how they should be used. 
```

Comment:
I think a separate extension spec is appropriate, given that a platform could choose not to use builders and instead work with stacks, lifecycles, and buildpacks directly.

Reply:
Incorporated